app/doi.py

```diff
@@ -0,0 +1,57 @@
+import os
+import json
+
+import requests
+from requests.exceptions import HTTPError
+
+import logging
+
+logger = logging.getLogger()
+
+
+def mint_doi(event, _context, _kwargs):
+    payload = event["body"]
```

Comment:
We can get away with this in prototype mode, but this is a point where we will need to validate input coming in from outside of our system. (The call will probably be generated with our sdk, but it might not be.)

Reply:
we're going to add the garden sdk as a dependency at some point, right? Once the other doi PR is up, validating could just be like: 
``` python
# from garden_ai.datacite import DataciteSchema
DataciteSchema.parse_obj(payload)
```

Reply:
Interesting! I suppose so. I definitely like the idea of having a library do the validation for us. Especially if it can spit out a nice error message. ("Expected a string here, got a ...")

Reply:
yeah that's a big selling point for pydantic, they've even got a CLI to automatically generate the appropriate pydantic classes for a target jsonschema (which is what I did w/ for datacite)